fluentinterface/index.md

```diff
@@ -0,0 +1,24 @@
+---
+layout: pattern
+title: Fluent Interface
+folder: fluentinterface
+permalink: /patterns/fluentinterface/
+categories: Architectural
+tags: Java
+---
+
+**Intent:** A fluent interface provides an easy-readable, flowing interface, that often mimics a domain specific language. This is especially useful when the users of your API benefit from code, that can nearly be read as human language.
+
+Fluent interfaces are sometimes confused with the Builder pattern, because they share method chaining and a fluent usage. However, fluent interfaces are not primarily used to create objects.
+
+![Fluent Interface](./etc/fluentinterface.png "Fluent Interface")
+
+**Applicability:** Use the Fluent Interface pattern when
+
+* you provide an API that would benefit from a DSL-like usage
+* you have objects that are difficult to configure or use
+
+**Real world examples:**
```

Comment:
What do you want me to write under the credits section? FluentInterface from Guava is linked in the real world examples and the interface's class documentation states the inspiration from Guava.